### DISCUSSION TITLE
The listing is missing 'item_weight', 'item_length', 'item_width', 'item_height', 'item_weight_unit' even that it's nullable

### DISCUSSION THREAD

**Discussion by simonj** [etsy/open-api#1126] [Q&A]
Hi Support,

Not sure if this discussions is still active 😅 

Am trying to use the API to use the **createDraftListing** endpoint. but when I send this data

```json
{
    "quantity": "1",
    "title": "test",
    "description": "test",
    "price": "23",
    "who_made": "i_did",
    "when_made": "made_to_order",
    "taxonomy_id": 1590,
    "is_supply": true,
    "type": "physical",
    "production_partner_ids": [3355757],
    "shipping_profile_id": "208339220653"
}
```

Endpoint returns with 
```json
{
    "error": "Could not set a calculated shipping profile to the listing.  The listing is missing 'item_weight', 'item_length', 'item_width', 'item_height', 'item_weight_unit' or 'item_dimensions_unit'."
}
```

But API says it is nullable 🤔
<img width="1261" alt="Skærmbillede 2023-10-06 kl  16 26 29" src="https://github.com/etsy/open-api/assets/183790/0ee09e95-fe21-4cea-a389-286dedff3c41">

**Comment by Gareth064** [comment]
Hey, you have to take the documentation with a very large pinch of salt. Etsy can't provided details on every single combination of possible uses cases (their words).

In your instance, the assumption I would go with here is that you have provided a Shipping Profile ID for a Shipping Profile which is "calculated"? If so, then Etsy are telling you that you now need to provide those extra details. I suppose you can't calculate shipping for an item if there are not details on the item to calculate from.

If it isn't a calculated shipping profile then you're probably going to have to reach out to Etsy support for some guidance.

developer@etsy.com

**Reply by simonj** [reply]
Thanks a lot, @Gareth064 It makes sense. The type was calculated. 

Yeah it's a little bit hard API to work with
